File: HealthRecords.Server/Models/Database/Prescription.cs
using System.ComponentModel.DataAnnotations;
using HealthRecords.Server.Models.Enum;

namespace HealthRecords.Server.Models.Database;

public class Prescription {
    [Key]
    [Required]
    public int Id { get; set; }
    [StringLength(255)]
    public required string Name { get; set; }
    public required float Dosage { get; set; }
    public required DosageUnit DosageUnit { get; set; }
    public required bool DosagePerKilogram { get; set; }
    public required int Frequency { get; set; }
    public required FrequencyUnit FrequencyUnit { get; set; }
    public required int Duration { get; set; }
    public required DurationUnit DurationUnit { get; set; }
    
    // Relationships
    public required int PatientId { get; set; }
    public virtual required Patient Patient { get; set; }
}

File: HealthRecords.Web/src/types/prescription.ts
import { z } from 'zod';
import { DosageUnitEnum, DurationUnitEnum, FrequencyUnitEnum } from './enums';

export const PrescriptionSchema = z.object({
	id: z.number(),
	name: z.string(),
	dosage: z.number(),
	dosageUnit: DosageUnitEnum,
	dosagePerKilogram: z.boolean(),
	frequency: z.number(),
	frequencyUnit: FrequencyUnitEnum,
	duration: z.number(),
	durationUnit: DurationUnitEnum,
	patientId: z.number(),
});
export type PrescriptionType = z.infer<typeof PrescriptionSchema>;

export const PrescriptionListSchema = z.array(PrescriptionSchema);
export type PrescriptionListType = z.infer<typeof PrescriptionListSchema>;

export const CreatePrescriptionSchema = z.object({
	name: z.string(),
	dosage: z.number(),
	dosageUnit: DosageUnitEnum,
	dosagePerKilogram: z.boolean(),
	frequency: z.number(),
	frequencyUnit: FrequencyUnitEnum,
	duration: z.number(),
	durationUnit: DurationUnitEnum,
});
export type CreatePrescriptionType = z.infer<typeof CreatePrescriptionSchema>;

export const UpdatePrescriptionSchema = z.object({
	name: z.string().optional(),
	dosage: z.number().optional(),
	dosageUnit: DosageUnitEnum.optional(),
	dosagePerKilogram: z.boolean().optional(),
	frequency: z.number().optional(),
	frequencyUnit: FrequencyUnitEnum.optional(),
	duration: z.number().optional(),
	durationUnit: DurationUnitEnum.optional(),
});
export type UpdatePrescriptionType = z.infer<typeof UpdatePrescriptionSchema>;
